go/tools/builders/generate_test_main.go

```diff
@@ -235,7 +237,19 @@ func main() {
 		}
 	}
 	{{end}}
-	bzltestutil.RegisterTimeoutHandler()
+
+	testTimeout := os.Getenv("TEST_TIMEOUT")
+	if testTimeout != "" {
+		flag.Lookup("test.timeout").Value.Set(testTimeout+"s")
+		// If Bazel sends a SIGTERM because the test timed out, it sends it to all child processes. Because
+		// we set -test.timeout according to the TEST_TIMEOUT, we need to ignore the signal so the test has
+		// time to properly produce the output (e.g. stack trace). It will be killed by Bazel after the grace
+		// period (15s) expires.
+		// If TEST_TIMEOUT is not set (e.g., when the test binary is run by Delve for debugging), we don't
+		// ignore SIGTERM so it can be properly terminated.
+		signal.Ignore(syscall.SIGTERM)
```

Comment:
@sluongng @fmeum Maybe we should do this instead here?

```
	c := make(chan os.Signal, 1)
	signal.Notify(c, syscall.SIGTERM)
	go func() {
		<-c
	}()
```

Reply:
With the intent of this being resettable by `signal.Reset`? I like that.